File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
void InOrder(TreeNode *root, vector<int>& result) {

    if (root == nullptr) return;
    InOrder(root->left, result);
    result.push_back(root->val);
    InOrder(root->right, result);
}
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ans;
        InOrder(root, ans);
        return ans;
    }
};

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
class Solution {
    public void helper(TreeNode root, List<Integer> result){
        if(root == null) return;

        if(root.left == null && root.right == null){
            result.add(root.val);
            return;
        }
   

        helper(root.left, result);

        result.add(root.val);

        helper(root.right, result);

    }
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();

        helper(root, result);

        return result;
    }
}